plugin/othttp/transport_test.go

```diff
@@ -0,0 +1,77 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package othttp
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"go.opentelemetry.io/otel/api/core"
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/api/propagation"
+	"go.opentelemetry.io/otel/api/trace"
+	mocktrace "go.opentelemetry.io/otel/internal/trace"
+)
+
+func TestTransportBasics(t *testing.T) {
+	var id uint64
+	tracer := mocktrace.MockTracer{StartSpanID: &id}
+	content := []byte("Hello, world!")
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := propagation.ExtractHTTP(r.Context(), global.Propagators(), r.Header)
+		span := trace.RemoteSpanContextFromContext(ctx)
+		tgtID, err := core.SpanIDFromHex(fmt.Sprintf("%016x", id))
+		if err != nil {
+			t.Fatalf("Error converting id to SpanID: %s", err.Error())
+		}
+		if span.SpanID != tgtID {
+			t.Fatalf("testing remote SpanID: got %s, expected %s", span.SpanID, tgtID)
+		}
+		if _, err := w.Write(content); err != nil {
+			t.Fatal(err)
+		}
+	}))
+	defer ts.Close()
+
+	r, err := http.NewRequest(http.MethodGet, ts.URL, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tr := NewTransport(
+		http.DefaultTransport,
+		WithTracer(&tracer),
+	)
+
+	c := http.Client{Transport: tr}
```

Comment:
I wonder if this could (easily) make a testable-example style test that would serve as documentation for how to use the new Transport implementation. Someone less familiar with the http library in Go will probably need an example somewhere.

Reply:
Good call.  I've added an `ExampleNewTransport()` to the tests that presents a minimal usable example.